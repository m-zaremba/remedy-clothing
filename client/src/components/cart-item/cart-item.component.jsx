import React, { memo } from 'react';
import PropTypes from 'prop-types';
import {
  CartItemContainer,
  ItemImage,
  ItemDetailsContainer,
  ItemName,
  ItemPrice,
} from './cart-item.styles';

const CartItem = ({ item: { imageUrl, name, price, quantity } }) => {
  return (
    <CartItemContainer>
      <ItemImage src={imageUrl} alt="item" />
      <ItemDetailsContainer>
        <ItemName>{name}</ItemName>
        <ItemPrice className="price">
          {quantity} x ${price}
        </ItemPrice>
      </ItemDetailsContainer>
    </CartItemContainer>
  );
};

CartItem.propTypes = {
  item: PropTypes.shape({
    name: PropTypes.string,
    imageUrl: PropTypes.string,
    price: PropTypes.number,
    quantity: PropTypes.number,
  }).isRequired,
};

export default memo(CartItem);
