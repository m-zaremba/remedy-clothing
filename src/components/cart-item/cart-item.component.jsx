import React from 'react';
import './cart-item.styles.scss';
import PropTypes from 'prop-types';

const CartItem = ({ item: { imageUrl, name, price, quantity } }) => {
  return (
    <div className="cart-item">
      <img src={imageUrl} alt="item" />
      <div className="item-details">
        <span className="name">{name}</span>
        <span className="price">
          {quantity} x ${price}
        </span>
      </div>
    </div>
  );
};

CartItem.propTypes = {
  item: PropTypes.shape({
    name: PropTypes.string,
    imageUrl: PropTypes.string,
    price: PropTypes.number,
    quantity: PropTypes.number,
  }).isRequired,
};

export default CartItem;
