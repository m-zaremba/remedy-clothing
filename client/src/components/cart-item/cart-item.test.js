import React from 'react';
import { shallow } from 'enzyme';

import { CartItem } from './cart-item.component';

describe('testing CartItem component', () => {
  it('should render CartItem component', () => {
    const mockItem = {
      imageUrl: 'www.testURL.com',
      name: 'testItem',
      price: 10,
      quantity: 1,
    };
    expect(shallow(<CartItem item={mockItem} />)).toMatchSnapshot();
  });
});
