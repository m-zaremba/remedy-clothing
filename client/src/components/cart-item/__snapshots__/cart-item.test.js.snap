// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CartItem component should render CartItem component 1`] = `ShallowWrapper {}`;
